inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -0,0 +1,557 @@
+@EntityScan({ "de.tudarmstadt.ukp.clarin.webanno.model",
+        "de.tudarmstadt.ukp.inception.search.model",
+        "de.tudarmstadt.ukp.clarin.webanno.security.model" })
+@TestPropertySource(locations = "classpath:MtasDocumentIndexTest.properties")
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@DataJpaTest
+@Transactional
+public class MtasDocumentIndexTest
+{
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    // Number of miliseconds to wait for the indexing to finish. This time must be enough
+    // to allow the index be built before the query is made. Otherwise, it could affect the
+    // test results. If this happens, a largest value could allow the test to pass.
+    private final int WAIT_TIME = 3000;
+
+    private final String temporaryFolderPath = "target/MtasDocumentIndexTest";
+
+    private @Autowired UserDao userRepository;
+
+    private @Autowired ProjectService projectService;
+    private @Autowired DocumentService documentService;
+    private @Autowired SearchService searchService;
+    private @Autowired AnnotationSchemaService annotationSchemaService;
+    private @Autowired ImportExportService importExportService;
+
+    @Before
+    public void setUp()
+    {
+        userRepository.create(new User("admin", Role.ROLE_ADMIN));
+
+        FileSystemUtils.deleteRecursively(new File(temporaryFolderPath));
+    }
+
+    @Test
+    public void testSimpleQuery() throws Exception
+    {
+        Project project = new Project();
+        project.setName("TestSimpleQuery");
+
+        Waiter createProjectWaiter = new Waiter();
+
+        // Start thread to create and initialize project
+        new Thread(() -> {
+            try {
+                projectService.createProject(project);
+                annotationSchemaService.initializeProject(project);
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+            createProjectWaiter.assertTrue(true);
+            createProjectWaiter.resume();
+        }).start();
+
+        // Wait for thread to complete
+        createProjectWaiter.await();
+
+        SourceDocument sourceDocument = new SourceDocument();
+
+        sourceDocument.setName("test");
+        sourceDocument.setProject(project);
+        sourceDocument.setFormat("text");
+
+        String fileContent = "The capital of Galicia is Santiago de Compostela.";
+
+        InputStream fileStream = new ByteArrayInputStream(
+                fileContent.getBytes(StandardCharsets.UTF_8));
+
+        Waiter uploadWaiter = new Waiter();
+
+        // Start thread to upload the document
+        new Thread(() -> {
+            try {
+                documentService.uploadSourceDocument(fileStream, sourceDocument);
+            }
+            catch (UIMAException | IOException e) {
+                e.printStackTrace();
+            }
+
+            try {
+                // Wait some time so that the document can be indexed
```

Comment:
Why dont you move this into the try/catch above?

Reply:
OK.